internal/services/consumption/consumption_budget_management_group_resource_test.go

```diff
@@ -299,16 +299,6 @@ resource "azurerm_consumption_budget_management_group" "test" {
         "baz",
       ]
     }
-
-    not {
```

Comment:
If this is not properly supported should we not be deprecating the property to remove in 4.0?

Reply:
lets remove it from the tests, deprecate the property with a explaination/link why we are, and then remove them behind a 4.0 flag?

Reply:
Hi @katbyte, I have updated the code and document to deprecate the `not` property. Kindly help review it again.